File: go/api/VSim.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// VSim 对应 VSim.py。
type VSim struct {
	*session.ApiGroup
}

// NewVSim 创建 VSim API 分组。
func NewVSim(s *session.Session) *VSim {
	return &VSim{ApiGroup: session.NewApiGroup(s)}
}

// OperateswitchVsim 对应 vsim/operateswitch-vsim。
func (v *VSim) OperateswitchVsim() (map[string]interface{}, error) {
	return v.S.Get("vsim/operateswitch-vsim", nil, "api")
}

File: src/api/v_sim.rs
//! VSim API group (`api/VSim.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// VSim API group.
pub struct VSim<'a> {
    conn: &'a Connection,
}

impl<'a> VSim<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        VSim { conn }
    }

    /// `vsim/operateswitch-vsim`. Operate the VSim switch.
    pub fn operateswitch_vsim(&self) -> Result<Value> {
        get_value(self.conn, "api/vsim/operateswitch-vsim")
    }
}
